mtp_api/apps/prison/serializers.py

```diff
@@ -80,6 +87,37 @@ class Meta:
         )
 
 
+class PrisonerAccountBalanceSerializer(serializers.Serializer):
+    NOMIS_ACCOUNTS = {'cash', 'spends', 'savings'}
+    combined_account_balance = serializers.SerializerMethodField()
+
+    def get_combined_account_balance(self, prisoner_location: PrisonerLocation):
+        if prisoner_location.prison.private_estate:
+            # NB: balances are not known in private estate currently
+            return 0
+
+        try:
+            nomis_account_balances = nomis.get_account_balances(
+                prisoner_location.prison.nomis_id,
+                prisoner_location.prisoner_number,
+            )
+            assert set(nomis_account_balances.keys()) == self.NOMIS_ACCOUNTS, 'response keys differ from expected'
+            assert all(
+                isinstance(nomis_account_balances[account], int) and nomis_account_balances[account] >= 0
+                for account in self.NOMIS_ACCOUNTS
+            ), 'not all response values are natural ints'
+        except AssertionError as e:
+            msg = f'NOMIS balances for {prisoner_location.prisoner_number} is malformed: {e}'
```

Comment:
Might be worth putting the whole `nomis_account_balances` data structure in the message to make it easier to debug

Reply:
yes i considered that, but we're supposed to keep our logs as clean of private information as possible. i figure that someone debugging would have the prisoner number so would actually make the call to NOMIS manually to see what was happening.